src/components/patterns/headingList.tsx

```diff
@@ -82,13 +83,47 @@ const HeadingList = () => {
 
       h2Elements.forEach(heading => observer.observe(heading))
 
+      // Handle initial anchor link on page load
+      const handleInitialAnchor = () => {
+        const hash = window.location.hash.slice(1)
+        if (hash) {
+          const targetElement = document.getElementById(hash)
+          if (targetElement) {
+            // Small delay to ensure layout is complete
+            setTimeout(() => {
+              targetElement.scrollIntoView({behavior: "smooth", block: "start"})
+              setActiveHeading(hash)
+            }, 100)
+          }
+        }
+      }
+
+      // Check for anchor on initial load
+      handleInitialAnchor()
+
       return () => {
         clearTimeout(timeoutId)
         observer.disconnect()
       }
     }, 250)
   }, [uuid])
 
+  // Handle hash changes with useEventListener
+  const handleHashChange = useCallback(() => {
```

Comment:
Now you have two functions doing the same thing. Can't we call this from within the hook?